### DISCUSSION TITLE
[@types/langs]: 'local' and 'name' are not assignable to parameter of type 'Type'.

### DISCUSSION THREAD

**Discussion by recursive-beast** [DefinitelyTyped/DefinitelyTyped#55386] [Issues with a @types package] [answered]
In the package's [documentation](https://www.npmjs.com/package/langs#documentation), this snippet is correct:

```node
import langs from "langs";

const local = "Français";
const language = langs.where("local", text);

console.log(language);

//  {
//    '1': 'fr',
//    '2': 'fra',
//    '3': 'fra',
//    'name': 'French',
//    'local': 'Français',
//    '2T': 'fra',
//    '2B': 'fre'
//  }
```

However, `langs.where` and `langs.has` accepts only these values as valid types for the first parameter:

```typescript
type Type = 1 | 2 | 3 | '1' | '2' | '2T' | '2B' | '3';
```

Here's the corresponding [line](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/45437e4d04961327656e72cc8f53e231a670353d/types/langs/index.d.ts#L7) in the declaration file.

[a comment by an automated account is not shown]

**Comment by AxelTerizaki** [comment]
I just ran into this very issue today. Is it possible to escalate this from a discussion to issue?

**Comment by recursive-beast** [comment]
I have just opened an issue #68508 for this, and I will start working on a PR.

**Comment by recursive-beast** [comment]
I issued a PR #68515 and I'm waiting for approval.

**Comment by recursive-beast** [comment, marked as answer]
PR #68515 merged.

@AxelTerizaki can you test on your end to see if everything works as expected ?

**Comment by AxelTerizaki** [comment]
Saw that :) tested it and it works. Thanks for the quick fix!

**Comment by recursive-beast** [comment]
Glad I can help 😄 I'll close the discussion for now.
